File: Source/GraphAlgorithms/num_connected_components.c
#include <stdio.h>

#define MAX 100

int visited[MAX];
int matrix[MAX][MAX];

void DFS(int v, int n) {
    visited[v] = 1;
    for(int i = 0; i < n; i++) {
        if(matrix[v][i] == 1 && visited[i] == 0) {
            DFS(i, n);
        }
    }
}

int numberOfConnectedComponents(int n) {
    int components = 0;
    for(int i = 0; i < n; i++) {
        if(visited[i] == 0) {
            DFS(i, n);
            components++;
        }
    }
    return components;
}

File: Source/GraphAlgorithms/num_connected_components.js
let visited = [];

function DFS(v, adjList) {
    visited[v] = true;
    for(let i = 0; i < adjList[v].length; i++) {
        if(!visited[adjList[v][i]]) {
            DFS(adjList[v][i], adjList);
        }
    }
}

function numberOfConnectedComponents(adjList) {
    let components = 0;
    for(let i = 0; i < adjList.length; i++) {
        if(!visited[i]) {
            DFS(i, adjList);
            components++;
        }
    }
    return components;
}
